### ISSUE TITLE
Errors reported by no-useless-escape have invalid locations on multiline template literals

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7643]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.2.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
rules:
  no-useless-escape: error
parserOptions:
  ecmaVersion: 6
```

**What did you do? Please include the actual source code causing the issue.**

```js
`this is a multiline template literal with
a useless escape \ character`;
```

**What did you expect to happen?**

```
2:18  error  Unnecessary escape character: \   no-useless-escape
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
1:60  error  Unnecessary escape character: \   no-useless-escape
```

**Comment by not-an-aardvark** [comment]
While fixing this bug, I noticed another issue: The locations of reported errors in template literals are all off by one.

```yml
rules:
  no-useless-escape: error
parserOptions:
  ecmaVersion: 6
```

```js
`useless\ escape`;
```

```bash
eslint no-useless-escape-test.js --format codeframe
```

**What did you expect to happen?**

```
error: Unnecessary escape character: \  (no-useless-escape) at no-useless-escape-test.js:1:8:
> 1 | `useless\ escape`;
    |        ^
  2 |
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
error: Unnecessary escape character: \  (no-useless-escape) at no-useless-escape-test.js:1:8:
> 1 | `useless\ escape`;
    |         ^
  2 |
```

**Comment by platinumazure** [comment]
Can we quote the cited useless escape while we're at it? Just having `\ ` (unquoted) makes it hard to tell what the problem is.
